js/type.js

```diff
@@ -1,590 +1,537 @@
-					
-				}
-			
-			}
-			else setText(ele, origHTML);
-			
-		}
-		
-		var adjustWidows = function () {
-		
-			for (i in rules) {
-			
-				var rule = rules[i];
-				
-				if (rule) {
-				
-					var	eles = document.querySelectorAll(rule.selector),
-							elescount = eles.length,
-							
-							method = fixMethod(rule.value);
-					
-					switch (method) {
-						
-						case 'paddingRight': case 'nonBreakingSpace': case 'paddingLeft': case 'wordSpacing': case 'letterSpacing' : case undefined :
-						
-							while (elescount-- > 0) {
-							
-								runTamer(eles[elescount], method);
-								
-							}
-							
-							break;
-						
-						default : console.log('Invalid method. Please use either padding-right, padding-left, word-spacing, or letter-spacing.'); 
-					
-					}
-				
-				}
-			
-			}
-		
-		};
-		
-		window.addEventListener('load', adjustWidows, false);
-		
-	},
-	
-	getStyle : function (ele, style, unit) {
-		
-		var ret;
-		
-		if (ele.currentStyle) ret = ele.currentStyle[style.replace(/-([A-z])/gi, function(a,b) {return b.toUpperCase();})];
-		else if (window.getComputedStyle) ret = document.defaultView.getComputedStyle(ele,null).getPropertyValue(style);
-		
-		if (unit) return parseFloat(ret);
-		else return ret;
-		
-	}
+  kerningPairs: function (rules) {
+
+    var traverseNodes = function (node, pairs) {
+      var next;
+
+      if (node.nodeType === 1) {
+        if (node = node.firstChild) {
+          do {
+            next = node.nextSibling;
+            traverseNodes(node, pairs);
+          } while (node = next);
+        }
+      }
+      else if (node.nodeType === 3) {
+        kernText(node, pairs);
+      }
+    };
+
+    var kernText = function (node, pairs) {
+
+      var nodetext = node.textContent,
+        nodechars = nodetext.split(''),
+        parent = node.parentNode,
+        pcount = pairs.length;
+
+      while (pcount-- > 0) {
+
+        var pair = pairs[pcount].replace(/^([\s\n\r\t]+|\/\*.*?\*\/)+/, '').replace(/[\s\n\r\t]+$/, ''),
+          chars = pair.match(/^(.)(.)\s/i),
+          amount = pair.match(/\s(\-*[0-9.]+[a-z]+)$/i)[1],
+          ccount = nodechars.length;
+
+        while (ccount-- > 0) {
+
+          var char = nodechars[ccount],
+            nextchar = nodechars[ccount + 1],
+            charpair = char + nextchar,
+            nextcharreg = new RegExp('^(<span[^>]+>)*' + chars[2] + '(<\/\s*span\s*>)*$', 'i');
+
+          if (char === chars[1] && nextchar && nextchar.match(nextcharreg)) {
+            nodechars[ccount] = '<span style="letter-spacing:' + amount + '">' + char + '</span>';
+          }
+        }
+      }
+
+      var temp = document.createElement('div');
+
+      temp.innerHTML = nodechars.join('');
+
+      while (temp.firstChild) {
+        parent.insertBefore(temp.firstChild, node);
+      }
+
+      parent.removeChild(node);
+    };
+
+    var kernAll = function () {
+
+      for (var i in rules) {
```

Comment:
I read in one of the closed issues that browser support is IE9+. I would then suggest moving these for...in loops on the array to Array.forEach.